Neo4j.Driver/Neo4j.Driver.Tests/Result/PeekingEnumeratorTests.cs

```diff
@@ -115,15 +115,32 @@ public void ShouldConsumeAllItemsAfterDiscard()
         public class DisposeMethod
         {
             [Fact]
-            public void ShouldNotDiscardAfterDispose()
+            public void ShouldDiscardAfterDispose()
             {
                 var list = new List<string> { "1", "2", "3" }.GetEnumerator();
                 var enumerator = new PeekingEnumerator<string>(list);
-                enumerator.MoveNext().Should().BeTrue();
-                enumerator.Consume();
+                enumerator.Dispose();
                 enumerator.MoveNext().Should().BeFalse();
                 enumerator.Peek().Should().BeNull();
-                enumerator.Position.Should().Be(3);
+                enumerator.Position.Should().Be(-1);
+            }
+
+            [Fact]
+            public void PeakShouldReturnDefaultAfterDispose()

```

Comment:
I do not think `Peek` should return something different from `Current`
The `Current` method will return `null` right now, see code [here](https://github.com/neo4j/neo4j-dotnet-driver/blob/1.1/Neo4j.Driver/Neo4j.Driver/Internal/Result/PeekingEnumerator.cs#L58)

Right now we do not have the problem of using `null` or `default(T)` as we actually use `list<object>`. But if we changed to `default(T)`, then I was worried that we might have some problems for `int` if we changed to `list<int>` or `list<dynamic>` or `list<T>` in the future. If we still use `null`, at least we got the option to use `int?` but if we return 0, then it might result in a strange bug in the end.

So I do not think this change is needed for now.